lib/common/mainloop.c

```diff
@@ -509,6 +509,59 @@ gio_poll_destroy(gpointer data)
     }
 }
 
+/*!
+ * \internal
+ * \brief Convert libqb's poll priority into GLib's one
+ *
+ * \param[in] prio  libqb's poll priority
+ *
+ * \return  best matching GLib's priority
+ */
+static gint
+conv_prio_libqb2glib(enum qb_loop_priority prio) {
+    gint ret = G_PRIORITY_DEFAULT;
+    switch (prio) {
+        case QB_LOOP_LOW:
+            ret = G_PRIORITY_LOW;
+            break;
+        case QB_LOOP_HIGH:
+            ret = G_PRIORITY_HIGH;
+            break;
+        default:
+            crm_trace("Invalid libqb's loop priority %d, assuming QB_LOOP_MED",
+                      prio);
+            break;
+    }
+    return ret;
+}
+
+/*!
+ * \internal
+ * \brief Convert libqb's poll priority to rate limiting spec
+ *
+ * \param[in] prio  libqb's poll priority
+ *
+ * \return  best matching rate limiting spec
+ */
+static enum qb_ipcs_rate_limit
+conv_libqb_prio2ratelimit(enum qb_loop_priority prio) {
+    /* this is an inversion of what libqb's qb_ipcs_request_rate_limit does */
+    enum qb_ipcs_rate_limit ret = QB_IPCS_RATE_NORMAL;
+    switch (prio) {
+        case QB_LOOP_LOW:
+            ret = QB_IPCS_RATE_SLOW;
+            break;
+        case QB_LOOP_HIGH:
+            ret = QB_IPCS_RATE_FAST;
+            break;
+        default:
+            crm_trace("Invalid libqb's loop priority %d, assuming QB_LOOP_MED",
+                      prio);
```

Comment:
Let's explicitly case QB_LOOP_MED b/c it's not invalid

Reply:
In the context of use, it was just fine, but you are right that might
be used in a more generic context.